packages/create-sitecore-jss/src/templates/nextjs-sxa/src/pages/404.tsx

```diff
@@ -0,0 +1,46 @@
+import { sitecorePagePropsFactory } from 'lib/page-props-factory';
+import Layout from 'src/Layout';
+import config from 'temp/config';
+import {
+  ComponentPropsContext,
+  GraphQLErrorPagesService,
+  SitecoreContext,
+} from '@sitecore-jss/sitecore-jss-nextjs';
+import { componentFactory } from 'temp/componentFactory';
+import { SitecorePageProps } from 'lib/page-props';
+import NotFound from 'src/NotFound';
+
+const Custom404 = (props: SitecorePageProps): JSX.Element => {
+  if (!(props && props.layoutData)) {
+    return <NotFound />;
+  }
+
+  return (
+    <ComponentPropsContext value={{}}>
+      <SitecoreContext componentFactory={componentFactory} layoutData={props.layoutData}>
+        <Layout layoutData={props.layoutData} headLinks={props.headLinks} />
+      </SitecoreContext>
+    </ComponentPropsContext>
+  );
+};
+
+export async function getStaticProps(context: SitecorePageProps) {
+  const props = await sitecorePagePropsFactory.create(context);
+  const errorPagesService = new GraphQLErrorPagesService({
+    endpoint: config.graphQLEndpoint,
+    apiKey: config.sitecoreApiKey,
+    siteName: props.site.name,
+    language: (props.site.locale || context.locale) as string,
```

Comment:
Just use `props.locale` (already uses `context.locale ?? props.site.language` which is what we want)
```suggestion
    language: props.locale,
```